independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Beans.java

```diff
@@ -274,7 +278,11 @@ static BeanInfo createProducerMethod(MethodInfo producerMethod, BeanInfo declari
             }
             if (DotNames.ALTERNATIVE_PRIORITY.equals(annotation.name())) {
                 isAlternative = true;
-                alternativePriority = annotation.value().asInt();
+                priority = annotation.value().asInt();
+                continue;
+            }
+            if (priority == null && DotNames.ARC_PRIORITY.equals(annotation.name())) {
```

Comment:
For future proofing and for annotation transformations, maybe add support for `@Priority` here (and below for fields) already?

Reply:
That's a good idea.